apps/music_proxy.cpp

```diff
@@ -205,48 +205,31 @@ class SteeringHandler
     {
         switch( _state )
         {
-        case monsteer::steering::SimulationPlaybackState::PLAY:
+        case monsteer::steering::State_PLAY:
         {
             _currentTime = musicTime;
             while( _subscriber.receive( 0 ))
              ;
             break;
         }
-        case monsteer::steering::SimulationPlaybackState::PAUSE:
+        case monsteer::steering::State_PAUSE:
         {
             _subscriber.receive( -1 );
             break;
         }
         }
     }
 
-    monsteer::steering::SimulationPlaybackState::State getPlaybackState() const
+    monsteer::steering::State getPlaybackState() const
     {
         return _state;
     }
 
-    std::string deserializeJSON( const zeroeq::FBEvent& event,
-                                 const std::string& schema )
-    {
-         flatbuffers::Parser parser;
-         if( !parser.Parse( schema.c_str( )))
-            throw std::runtime_error( parser.error_ );
-
-         std::string json;
-         flatbuffers::GeneratorOptions opts;
-         opts.base64_byte_array = true;
-         opts.strict_json = true;
-         GenerateText( parser, event.getData(), opts, &json );
-         return json;
-    }
-
 private:
 
-    void _onStimulusInjection( const zeroeq::FBEvent& event )
+    void _onStimulusInjection( monsteer::steering::ConstStimulusInjectionPtr event )
     {
-        LBASSERT( event.getTypeIdentifier() == monsteer::steering::EVENT_STIMULUSINJECTION )
-        const std::string& json = deserializeJSON( event,
-                                monsteer::steering::SCHEMA_STIMULUSINJECTION );
+        const std::string& json = event->toJSON();
```

Comment:
The function toJSON produces a JSON string that includes only the event data or also the event type?
If the event type is part of the JSON string, then we can remove the messageID attribute from the fbs file and adapt the Python code accordingly.